icu4c/source/common/rbbiscan.cpp

```diff
@@ -1269,6 +1269,7 @@ void RBBIRuleScanner::scanSet() {
 
         n = pushNewNode(RBBINode::setRef);
         if (U_FAILURE(*fRB->fStatus)) {
+            delete uset;
```

Comment:
Optional: change uset to be a LocalPointer<UnicodeSet>, and get rid of all of the explicit deletes of it on error conditions within this function. This would be cleaner and less error prone, but is a more intrusive change.